### ISSUE TITLE
50 Minute Token Refresh is Not Working

### ISSUE THREAD

**Issue by zjullion** [aws-amplify/amplify-js#6343]
**Describe the bug**
https://github.com/aws-amplify/amplify-js/pull/4205 is not working - tokens should be automatically refreshed once they have 10 min or less to expire, but this is not happening.

**To Reproduce**
Open an `amplify-js` application (with cognito authentication), wait for 55 min, then call `const session = await Auth.currentSession()` and see that `session.accessToken.payload.exp` is less than 300 - with each repeated call you will see this value decrease, and the token will not be refreshed.

**Expected behavior**
I expect tokens to refresh when they have less than 10 min to expire.

**Code Snippet**
```
    const session = await Auth.currentSession()
    const remainingTime = session.accessToken.payload.exp - (new Date().getTime() / 1000)
    console.log(remainingTime)
```

I should note that the following snippet works just fine (and we're writing custom code now to check `session.accessToken.payload.exp` and run this snippet when there is little time left):
```
        user.refreshSession(session?.refreshToken, (error) => {
          if (error) {
            console.log('failed to refresh token')
          }
        })
```

<details>
  <summary><strong>Environment</strong></summary>

```
  System:
    OS: macOS 10.15.4
    CPU: (8) x64 Intel(R) Core(TM) i7-4980HQ CPU @ 2.80GHz
    Memory: 281.47 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 12.18.0 - ~/.nvm/versions/node/v12.18.0/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.4 - ~/.nvm/versions/node/v12.18.0/bin/npm
  Browsers:
    Chrome: 83.0.4103.116
    Firefox: 78.0.1
    Safari: 13.1
  npmPackages:
    @amaabca/react-components: ^14.9.0 => 14.9.0
    @amaabca/sensitive-param-filter: ^1.0.6 => 1.0.6
    @babel/cli: ^7.2.3 => 7.10.1
    @babel/plugin-transform-runtime: ^7.5.5 => 7.10.1
    @datadog/browser-rum: ^1.5.0 => 1.12.1
    @testing-library/cypress: ^6.0.0 => 6.0.0
    @testing-library/jest-dom: ^5.1.1 => 5.9.0
    @testing-library/react: ^9.4.0 => 9.5.0
    @testing-library/user-event: ^10.3.1 => 10.4.1
    @types/lodash.orderby: ^4.6.6 => 4.6.6
    @welldone-software/why-did-you-render: ^4.0.5 => 4.2.5
    apollo-link: ^1.2.11 => 1.2.14
    aws-amplify: ^3.0.18 => 3.0.18
    aws-amplify-react: ^3.1.2 => 3.1.9
    aws-appsync: ^3.0.2 => 3.0.3
    aws-sdk: ^2.374.0 => 2.690.0
    axios: ^0.18.0 => 0.18.1
    axios-mock-adapter: ^1.18.1 => 1.18.1
    babel-preset-react: ^6.24.1 => 6.24.1
    commander: ^4.1.0 => 4.1.1
    currency.js: ^1.2.1 => 1.2.2
    cypress: 4.10.0 => 4.10.0
    danger: ^7.0.1 => 7.1.4
    deep-equal: ^2.0.1 => 2.0.3
    deepcopy: ^2.0.0 => 2.0.0
    enzyme: ^3.9.0 => 3.11.0
    enzyme-adapter-react-16: ^1.11.2 => 1.15.2
    enzyme-to-json: ^3.3.5 => 3.5.0
    esm: ^3.2.25 => 3.2.25
    fuse.js: ^3.4.5 => 3.6.1
    glob: ^7.1.6 => 7.1.6
    graphql-tag: ^2.10.1 => 2.10.3
    husky: ^1.3.1 => 1.3.1
    idle-js: ^1.2.0 => 1.2.0
    inquirer: ^6.2.1 => 6.5.2
    isomorphic-fetch: ^2.2.1 => 2.2.1
    jest-styled-components: ^7.0.2 => 7.0.2
    jest-when: ^2.3.1 => 2.7.2
    lodash.orderby: ^4.6.0 => 4.6.0
    mock-fs: ^4.10.1 => 4.12.0
    mock-spawn: ^0.2.6 => 0.2.6
    mockdate: ^2.0.5 => 2.0.5
    moment: 2.25.3 => 2.25.3
    moment-timezone: 0.5.28 => 0.5.28
    node-sass: ^4.12.0 => 4.14.1
    nvmrc-check: ^1.0.2 => 1.0.2
    prop-types: ^15.7.2 => 15.7.2
    query-string: ^6.4.0 => 6.12.1
    random-name: ^0.1.2 => 0.1.2
    react: ^16.12.0 => 16.13.1
    react-dom: ^16.12.0 => 16.13.1
    react-hotkeys: ^2.0.0 => 2.0.0
    react-joyride: ^2.2.1 => 2.2.1
    react-router-dom: ^5.0.0 => 5.2.0
    react-scripts: 3.3.0 => 3.3.0
    react-select: ^2.1.2 => 2.4.4
    react-testing-library: 6.1.2 => 6.1.2
    regenerator-runtime: ^0.13.2 => 0.13.5
    request-promise-native: ^1.0.8 => 1.0.8
    rollbar: ^2.11.0 => 2.16.2
    snazzy: ^8.0.0 => 8.0.0
    standard: ^12.0.1 => 12.0.1
    striptags: ^3.1.1 => 3.1.1
    styled-components: ^5.0.0 => 5.1.1
    use-deep-compare-effect: ^1.3.1 => 1.3.1
    uuid: ^3.3.2 => 3.4.0
    whatwg-fetch: ^3.0.0 => 3.0.0
  npmGlobalPackages:
    npm: 6.14.4
```

</details>

**Comment by zjullion** [comment]
Okay, after making some changes to our application, I am now very certain something is broken with the way `amplify` is refreshing tokens. We run code that looks like this before every graphQL call:

```
    const session = await Auth.currentSession()
    const user = await Auth.currentAuthenticatedUser()
    const remainingTime = session?.accessToken?.payload?.exp - (new Date().getTime() / 1000)
    console.log(`${remainingTime} seconds before token expires`)
```

And we just received a log with `-257.85700011253357 seconds before token expires`. This token was almost five minutes expired, and both `Auth.currentSession()` and `Auth.currentAuthenticatedUser()` returned without throwing any error!

**Comment by disrael** [comment]
We at [Uclusion](https://www.uclusion.com) really need this.

**Comment by wlee221** [comment]
Hi, this should be fixed by #7288. Please let us know if the issues persists!
